### PR TITLE
Upgrade to ring-0.16

### PR DESCRIPTION
**Pull request by rushidesai**
`extract_and_expand` has gone from `ring`. So I had to create some utility functions for this. Not sure where to put them.

### CODE REVIEW

File: src/transport/mod.rs [hunk 1]
```diff
@@ -1,4 +1,28 @@
...
+
+/// Generic newtype wrapper that lets us implement traits for externally-defined
+/// types.
+#[derive(Debug, PartialEq)]
+struct WrappedBuffer([u8; 32]);
+
+impl From<hkdf::Okm<'_, hkdf::Algorithm>> for WrappedBuffer {
+    fn from(okm: hkdf::Okm<hkdf::Algorithm>) -> Self {
+        let mut r = [0u8; 32];
+        okm.fill(&mut r).unwrap();
```
Comment (ewilken) [inline src/transport/mod.rs:25]: This panicked when I tested the pairing process.
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Unspecified', src/transport/mod.rs:25:9
```
Maybe it should rather be a `TryFrom`?

**Review by ewilken — CHANGES_REQUESTED** [review]
Thanks for your contribution! :)

Unfortunately, your `WrappedBuffer` implementation seems to panic on my machine.

Also, the formatting check in CI is complaining. I'm using the nightly `rustfmt` on this project, so a `cargo +nightly fmt` should do the trick.

**Comment by DaleLJefferson** [comment]
I would love to see this PR merged, i'm seeing the same panic.

I'm trying to use `hap` with `rumqttc` and without this PR (upgrade Ring) it does not compile (ring version conflict). 0.13 vs 0.16.

**Comment by ewilken** [comment]
I just released `v0.1.0-pre.4` that removes the dependency on `ring` entirely in favor of RustCrypto's `hkdf` crate. So this PR is obsoleted and your use case should work now. So closing.

Thanks a lot for caring! Let me know when I can provide any further help!

**Pull request closed without merging**